packages/block-editor/src/components/block-tools/block-contextual-toolbar.js

```diff
@@ -7,17 +7,68 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { forwardRef, useEffect, useRef, useState } from '@wordpress/element';
 import { hasBlockSupport, store as blocksStore } from '@wordpress/blocks';
 import { useSelect } from '@wordpress/data';
+import { ToolbarItem, Button } from '@wordpress/components';
+import { levelUp } from '@wordpress/icons';
+import { useViewportMatch } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import NavigableToolbar from '../navigable-toolbar';
 import BlockToolbar from '../block-toolbar';
 import { store as blockEditorStore } from '../../store';
+import BlockIcon from '../block-icon';
+
+const CollapseFixedToolbarButton = forwardRef( ( { onClick }, ref ) => {
+	return (
+		<ToolbarItem
+			as={ Button }
+			className="components-button components-toolbar-button block-editor-block-toolbar__collapse-fixed-toolbar"
```

Comment:
Used `ToolbarButton` with success in [6663a49](https://github.com/WordPress/gutenberg/pull/49634/commits/6663a499f3bf6cc528bdc68f250afaf6dcd1f7e3), thank you!

Reply:
> where we want X to always respond to updates in styling across the base component system, it is bad to reuse classnames

That is because our system is components based, and not classname-based — ie. the public APIs of the package are the React Components.

> It's the core product using a private API, not using a private API from an extending product :)

From the point of view of an external consumer of the `@wordpress/components` package, those classnames should not exist (this applies also to other packages in Gutenberg like block editor).

Re-using such classnames across the editor just to "borrow" styles completely breaks the assumption of having a components system that abstracts semantics and design.

Reply:
Yea you are perfectly right and I understand what you say. I was a bit hot headed in the beginning of our conversation. ;))